data/src/main/java/com/example/pinterbest/data/repository/AuthRepositoryImpl.kt

```diff
@@ -0,0 +1,87 @@
+        emit(Result.Loading)
+        try {
+            val response = ApiClient().getInstance().getClient().getAuthCheck(
+                SessionRepositoryImpl(preferences).authProvider() ?: ""
+            )
+            if (response.isSuccessful) {
+                emit(Result.Success(response.code()))
+            } else {
+                if (ErrorMessage.ErrorMap[response.code()] != null) {
+                    emit(Result.Error(ErrorMessage.ErrorMap[response.code()]!!))
+                }
+            }
+        } catch (e: UnknownHostException) {
+            emit(Result.Error(UnknownHostException()))
+        }
+    }.flowOn(Dispatchers.IO)
+
+    override suspend fun postLogIn(userLogIn: UserLogIn):
+        Flow<Result<ResponseWithCookie>> = flow {
+            emit(Result.Loading)
+            try {
+                val response = ApiClient().getInstance().getClient().postLogIn(
+                    userData = userLogIn
+                )
+                if (response.isSuccessful) {
+                    val responseWithCookie = ResponseWithCookie(
+                        response.code(),
+                        response.headers().values("Set-Cookie")[0].substringBefore(";")
+                    )
+                    emit(Result.Success(responseWithCookie))
+                    saveCookieToPreferences(responseWithCookie.cookie)
+                } else {
+                    if (ErrorMessage.ErrorMap[response.code()] != null) {
+                        emit(Result.Error(ErrorMessage.ErrorMap[response.code()]!!))
+                    }
+                }
+            } catch (e: UnknownHostException) {
+                emit(Result.Error(UnknownHostException()))
+            }
+        }.flowOn(Dispatchers.IO)
+
+    override suspend fun postSignUp(userSignUp: UserSignUp):
+        Flow<Result<ResponseWithCookie>> = flow {
+            emit(Result.Loading)
+            try {
+                val response = ApiClient().getInstance().getClient().postSignUp(
+                    userData = userSignUp
+                )
+                if (response.isSuccessful) {
+                    val responseWithCookie = ResponseWithCookie(
+                        response.code(),
+                        response.headers().values("Set-Cookie")[0].substringBefore(";")
+                    )
+                    emit(Result.Success(responseWithCookie))
+                    saveCookieToPreferences(responseWithCookie.cookie)
+                } else {
+                    if (ErrorMessage.ErrorMap[response.code()] != null) {
+                        emit(Result.Error(ErrorMessage.ErrorMap[response.code()]!!))
+                    }
+                }
+            } catch (e: UnknownHostException) {
+                emit(Result.Error(UnknownHostException()))
+            }
+        }.flowOn(Dispatchers.IO)
+
+    override fun saveCookieToPreferences(cookie: String) {
+        SessionRepositoryImpl(preferences).saveSession(cookie)
```

Comment:
SessionRepositoryImpl must be provided outside of class via constructor. Creating any dependencies inside it's a bad practice

Reply:
Fixed